sdk/metric/processor/basic/basic.go

```diff
@@ -171,13 +163,17 @@ func (b *Processor) Process(accum export.Accumulation) error {
 		}
 		if stateful {
 			if desc.InstrumentKind().PrecomputedSum() {
-				// If we know we need to compute deltas, allocate two aggregators.
-				b.AggregatorFor(desc, &newValue.cumulative, &newValue.delta)
-			} else {
-				// In this case we are certain not to need a delta, only allocate
-				// a cumulative aggregator.
-				b.AggregatorFor(desc, &newValue.cumulative)
+				// To convert precomputed sums to
+				// deltas requires two aggregators to
+				// be allocated, one for the prior
+				// value and one for the output delta.
+				// This functionality was removed from
+				// the basic processor in PR #1415.
+				return aggregation.ErrNoCumulativeToDelta
```

Comment:
What do you do with this error in the caller? I think this should be returned to the user when setting the expected temporality instead of framework dealing with this error.

Reply:
The reason this isn't returned earlier is that the decision is passed as a function to allow per-instrument configuration. I guess the functionality here is fairly low-level, and if we get to a higher-level configuration mechanism that layer will have to verify something like this.  There's a `CumulativeAggregationTemporalitySelector` but not a corresponding `DeltaAggregationTemporalitySelector`, so the caller has to go out of their way to configure this condition.